tests/e2e/gc_rwx_multi_ns_gc.go

```diff
@@ -0,0 +1,293 @@
+
+		ginkgo.By("Creating a PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteMany)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client,
+			[]*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		pvcNameInSV := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(pvcNameInSV).NotTo(gomega.BeEmpty())
+		fcdIDInCNS := getVolumeIDFromSupervisorCluster(pvcNameInSV)
+		gomega.Expect(fcdIDInCNS).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			if pvclaim != nil {
+				err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(fcdIDInCNS)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		// Verify using CNS Query API if VolumeID retrieved from PV is present.
+		framework.Logf("Invoking QueryCNSVolumeWithResult with VolumeID: %s", fcdIDInCNS)
+		queryResult, err := e2eVSphere.queryCNSVolumeWithResult(fcdIDInCNS)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(queryResult.Volumes).ShouldNot(gomega.BeEmpty())
+		framework.Logf("volume Name:%s, capacity:%d volumeType:%s health:%s accesspoint: %s",
+			queryResult.Volumes[0].Name,
+			queryResult.Volumes[0].BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails).CapacityInMb,
+			queryResult.Volumes[0].VolumeType, queryResult.Volumes[0].HealthStatus,
+			queryResult.Volumes[0].BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails).AccessPoints)
+
+		if volHealthCheck {
+			ginkgo.By("poll for health status annotation")
+			err = pvcHealthAnnotationWatcher(ctx, client, pvclaim, healthStatusAccessible)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		// Changing the reclaim policy of the pv to retain.
+		ginkgo.By("Changing the volume reclaim policy")
+		persistentvolumes[0].Spec.PersistentVolumeReclaimPolicy = v1.PersistentVolumeReclaimRetain
+		pv, err = client.CoreV1().PersistentVolumes().Update(ctx, persistentvolumes[0],
+			metav1.UpdateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			if pv != nil {
+				ginkgo.By("Deleting the PV1")
+				err = client.CoreV1().PersistentVolumes().Delete(ctx, pv.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		//Add a check to validate CnsVolumeMetadata crd
+		verifyCRDInSupervisorWithWait(ctx, f, pvcNameInSV, crdCNSVolumeMetadatas, crdVersion, crdGroup, true)
```

Comment:
there will be two crds one for pvc and one for pv and we need to verify the entity references as well
we need to do the same in other places as well

Reply:
Replaced with waitAndVerifyCnsVolumeMetadata4GCVol helper method which does metadata validations